weather_stats/weatherstatsapi/weather/cache.py

```diff
@@ -0,0 +1,64 @@
+import logging
+import os
+from typing import Optional
+
+import redis.asyncio as redis
+
+from weather.schemas import Weather
+from weather.settings import APP_SETTINGS
+
+logger = logging.getLogger(os.path.basename(__file__))
+
+
+class RedisCache:
+    __client: Optional[redis.Redis] = None
+
+    @classmethod
+    async def __get_redis(cls) -> Optional[redis.Redis]:
+        if cls.__client is not None:
+            return cls.__client
+        try:
+            redis_client = await redis.Redis(
+                host=APP_SETTINGS.redis_host,
+                port=APP_SETTINGS.redis_port,
+                db=APP_SETTINGS.redis_db,
+                password=APP_SETTINGS.redis_password,
+            )
+            pong = await redis_client.ping()
+            if pong:
+                cls.__client = redis_client
+        except redis.RedisError:
+            logger.warning("Cannot connect to redis", exc_info=True)
+        return cls.__client
+
+    @classmethod
+    def __location_to_key(cls, latitude: float, longitude: float) -> str:
+        return f"weather.{latitude}.{longitude}"
+
+    @classmethod
+    async def is_connection_up(cls) -> bool:
```

Comment:
This method looks like it will return always `True` as `cls.__get_redis()` tries to connect to redis if it was not done previously. `False` would happen only od redis connection error. Looks like unintended side effect.